csharp/extractor/Semmle.Extraction.CSharp/Entities/Constructor.cs

```diff
@@ -33,68 +49,79 @@ public override void Populate(TextWriter trapFile)
         protected override void ExtractInitializers(TextWriter trapFile)
         {
             // Do not extract initializers for constructed types.
-            if (!IsSourceDeclaration)
+            // Only extract initializers for constructors with a body and primary constructors.
+            if (Block is null && ExpressionBody is null && !IsPrimary ||
+                !IsSourceDeclaration)
+            {
                 return;
+            }
```

Comment:
As discussed on Slack, this might also apply to default constructors too.
```
class Base {}
class Derived : Base {}
```
`Derived` has a default constructor, which implicitly is calling `base()`.

Reply:
same comment as above (and thank you for putting it here!)